### PR TITLE
feat: nextprot api binding

### PR DESCRIPTION
**Pull request by sachdeva-shrey**
Signed-off-by: sachdeva-shrey <shreysachdeva.2000@gmail.com>

## Changes
1. Integrate neXtProt API using SDK to visualise API data using `feature-viewer`.
2. Add isoform selection feature.
3. Fetches entry from URL param.

### CODE REVIEW

File: src/components/FeatureViewer.tsx [hunk 1]
```diff
@@ -1,53 +1,115 @@
...
-      });
+    let sequences: IsoformType[];
+    let features: FeatureData[];
 
-      setData(variantValues);
-    });
+    Promise.all(getFeaturesByView(featureList, nx, paramEntry!)).then(
+      (rawData) => {
+
+        sequences = rawData[0];
```
Comment (kwsamarasinghe) [inline src/components/FeatureViewer.tsx]: Null and empty check for rawData?

File: src/components/Table.tsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ const Table = (props: TableProps) => {
       data,
       getExportFileBlob,
       //@ts-ignore
-      initialState: { pageIndex: 0, pageSize: 5 },
+      initialState: { pageIndex: 0, pageSize: 10 },
```
Comment (kwsamarasinghe) [inline src/components/Table.tsx]: Can page size be in a config file?

File: src/utils/helpers/getFeatures.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  let featuresForViewer = [];
+  let featForViewer;
+  let isoLengths = {};
+
+  rawData.forEach((i: any) => {
+    //@ts-ignore
+    isoLengths[i.uniqueName] = i.sequenceLength;
+  });
+
+  for (let i = 1; i < rawData.length - 1; i++) {
```
Comment (kwsamarasinghe) [inline src/utils/helpers/getFeatures.ts:63]: Can't we only iterate through rawData once? Perhaps using an index variable?

File: src/utils/helpers/getFeatures.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+export function addFeatures(
+  featuresForViewer: FeatsForViewer[],
+  isoName: string
+) {
+  let featureList: FeatureData[] = [];
+  for (let i = 0; i < featuresForViewer.length; i++) {
+    if (
+      Object.keys(featuresForViewer[i]).length !== 0 &&
```
Comment (kwsamarasinghe) [inline src/utils/helpers/getFeatures.ts]: let currentFeature = featuresForViewer[i] and use the variable later

File: src/utils/helpers/getFeatures.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      featureList.push({ ...featuresForViewer[i][isoName] });
+    }
+  }
+  return featureList;
+}
+
+export function getMetadataByView() {
+  let metaData: MetaData[] = [];
+
+  for (let i = 0; i < featureList.length - 1; i++)
```
Comment (kwsamarasinghe) [inline src/utils/helpers/getFeatures.ts]: Better use a linter if you are not already using.

Comment (kwsamarasinghe) [inline src/utils/helpers/getFeatures.ts]: Why do you prefer the indexed for loop instead of featureList.map ? because you want to ignore the last element?

**Review by kwsamarasinghe — APPROVED** [review]
Please check my minor comment. Otherwise looks fine for me, please address them and merge yourself.

File: src/utils/helpers/getFeatures.ts (hunk 5 above)
Comment (sachdeva-shrey) [reply src/utils/helpers/getFeatures.ts]: We can use `map` here. 👍

Comment (sachdeva-shrey) [reply src/utils/helpers/getFeatures.ts]: I was using ESLint that comes pre-configured with `react-scripts`. I have however added pre-commit hooks and prettier config.

File: src/utils/helpers/getFeatures.ts (hunk 3 above)
Comment (sachdeva-shrey) [reply src/utils/helpers/getFeatures.ts:63]: We can actually remove the first iteration, I realised that it was redudant. Thanks for pointing out.

**Pull request merged**